node/task.ts

```diff
@@ -766,14 +872,73 @@ export function cp(source: string, dest: string, options?: string, continueOnErr
  * @param     continueOnError optional. whether to continue on error
  */
 export function mv(source: string, dest: string, options?: string, continueOnError?: boolean): void {
-    if (options) {
-        shell.mv(options, source, dest);
+    if (!options) {
+        options = '';
     }
     else {
-        shell.mv(source, dest);
+        options = options.toLowerCase();
     }
+    try {
+        if (!fs.existsSync(source)) {
+            throw new Error(loc('LIB_OperationFailed', source + ' doesnt exist'));
+        }
+        if (getPlatform() == Platform.Windows) {
+            if (fs.statSync(source).isDirectory()) {
+                mkdirP(dest);
+                let command: string = 'robocopy ' + source + ' ' + dest + ' /move';
```

Comment:
- robocopy retries 1 million times by default (too much)
- we should use tool runner
  - it handles quoting
  - it roots the tool (otherwise additional quoting problems on windows)
  - it avoids execSync (buffers output until tool completes - i.e. user perceives as hang)
- one drawback of robocopy is, it will produce garbled output in some cases. It does not follow the console code page. It follows the system default (i.e. 1252 on your box, instead of 437). We need to solve this, although I don't have a good suggestion offhand. We might want to bundle a little wrapper exe. Here is what I did in the [PulishBuildArtifacts task](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/Tasks/PublishBuildArtifactsV1/Invoke-Robocopy.ps1)

Reply:
I changed the retries and incorporated toolrunner.

Concerning the robocopy output, to be clear is this an issue with how the tool displays output to the console or how the files themselves get copied? It looked like the copy itself was working for me (I tested with 'ç').

I'm not really sure to what degree we care about garbled output from the tool itself.  I guess I'm just not seeing the use case where users would have the type of special characters that would be affected in file paths and be upset that they're not displaying correctly - obviously you've run into this before though, is there something I'm missing?

Reply:
After offline discussion changed this to line up with the original comment.